### ISSUE TITLE
exec parser should validate procedure name length

### ISSUE THREAD

**Issue by hackaugusto** [0xPolygonMiden/miden-vm#671]
Procedure names are limited to [100 chars](https://github.com/0xPolygonMiden/miden-vm/blob/49e8afcc11fd91169ff2755b6b63145f156fa241/assembly/src/procedures/mod.rs#L146). For both `proc` and `export` [ref](https://github.com/0xPolygonMiden/miden-vm/blob/49e8afcc11fd91169ff2755b6b63145f156fa241/assembly/src/tokens/mod.rs#L102-L122)

Exec's argument is a proc name, and its [validation](https://github.com/0xPolygonMiden/miden-vm/blob/49e8afcc11fd91169ff2755b6b63145f156fa241/assembly/src/tokens/mod.rs#L180) currently doesn't check the procedure's name length.

The above is not a serious issue, since later on the procedure name lookup will fail. The issue is only to make the parsing a bit more strict / consistent.

I wonder if it would also make sense to add this constraint to the module names.

**Comment by Danijel-Enoch** [comment]
Good Day, Is the project still active ? I would like to work on this issue

**Comment by frisitano** [comment]
Hi @Danijel-Enoch I wanted to make you aware of some work I am doing on #692.  I have refactored the naming validation code such that is applicable to assembly constants I am introducing.  I'm just letting you know because I don't want us to cross wires.

**Comment by Danijel-Enoch** [comment]
@frisitano OKay

**Comment by Danijel-Enoch** [comment]
@hackaugusto @bobbinth I am new to open source contribution so i might assistance

So I took a look at [ref link](https://github.com/0xPolygonMiden/miden-vm/blob/49e8afcc11fd91169ff2755b6b63145f156fa241/assembly/src/tokens/mod.rs#L180) made by @hackaugusto for where the validation is needed, so i saw that there is parse_exec function and a validate_proc_invocation_label function.

my question would the validation be a seperate function or would it be inside the parse_exec function or  inside the validate_proc_invocation_label function.

**Comment by frisitano** [comment]
@Danijel-Enoch if you take a look in the `validate_proc_invocation_label` function there is a function call to `is_valid_label`.  It is within `is_valid_label` that the naming conventions should be enforced.   In the assembly constants PR I am working on I have extracted the name validation logic to a function [`parse_name`](https://github.com/0xPolygonMiden/miden-vm/pull/692/files#diff-f9b5d54ca2c50bc386d26ee23b1c180b4f97fab64d7517a90a96c56b71264346R403).  I would propose you replace `is_valid_label` with `parse_name` and pass it the appropriate [name configuration](https://github.com/0xPolygonMiden/miden-vm/pull/692/files#diff-65db0a7cb6086e1d9ee1ae536cf7543a98a3a72d10717d8f06df4a71ecb8cff4R77).  It would probably be good to remove `is_valid_label` entirely and replace with `parse_name` everywhere (I may rename `parse_name` to `parse_label`).  It is also used when validating import paths [here](https://github.com/0xPolygonMiden/miden-vm/blob/next/assembly/src/tokens/mod.rs#L245).

**Comment by Danijel-Enoch** [comment]
@frisitano  I do not Understand what you mean by "pass it the appropiate [name configuration](https://github.com/0xPolygonMiden/miden-vm/pull/692/files#diff-65db0a7cb6086e1d9ee1ae536cf7543a98a3a72d10717d8f06df4a71ecb8cff4R77)".

**Comment by frisitano** [comment]
@Danijel-Enoch  So if you look at this function signature `pub(crate) fn parse_name(name: String, config: NameConfig) -> Result<String, NameError> ` it takes two arguments:
- name: String - the string that we want to validate
- config: NameConfig - A configuration struct that specifies the rules that the name should conform to.  

I linked you to the appropriate `NameConfig` which should be used as `config` when calling `parse_name(...)` to validate procedures. 
```
/// The name configuration for procedures.
const PROCEDURE_NAME_CONFIG: NameConfig = NameConfig {
    max_len: MAX_NAME_LEN,
    start_with_letter: true,
    numbers_letters_underscore: true,
    caps: false,
};
```

**Comment by Danijel-Enoch** [comment]
So I git cloned to my PC but i did not find the  `is_valid_label` anywhere in the code. Should i just  work from github/githubdev or is there a way around it

**Comment by frisitano** [comment]
@Danijel-Enoch You would need to checkout the appropriate branch.  We typically work from `next` so you should branch off of that.  Please take a look at the PR #692 to get some insights of how to use the new `LabelParser` for your implementation.

**Comment by tohrnii** [comment]
@Danijel-Enoch It might also be helpful to go through the [contribution guidelines](https://github.com/0xPolygonMiden/miden-vm/blob/main/CONTRIBUTING.md) once.

**Comment by bobbinth** [comment]
Closed by #838.
